examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2097,200 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
 
 	atomic.StoreInt32(&stop, 1)
 }
+
+// AggregationExamples contains examples of aggregation operations.
+func AggregationExamples(t *testing.T, db *mongo.Database) {
+	ctx := context.Background()
+
+	coll := db.Collection("sales")
+	coll1 := db.Collection("air_alliances")
+
+	err := coll.Drop(context.Background())
+	require.NoError(t, err)
+	err1 := coll1.Drop(context.Background())
+	require.NoError(t, err1)
+
+	{
+		// Start Aggregation Example 1
+		pipeline := bson.A{
+			bson.D{{"$match", bson.D{{"items.fruit", "banana"}}}},
+			bson.D{{"$sort", bson.D{{"date", 1}}}},
+		}
+
+		cs, err := coll.Aggregate(ctx, pipeline)
+		require.NoError(t, err)
+		defer cs.Close(ctx)
+
+		ok := cs.Next(ctx)
+		next := cs.Current
+
+		// End Aggregation Example 1
+
+		require.True(t, ok)
+		require.NoError(t, err)
+		require.NotEqual(t, len(next), 0)
```

Comment:
This assertion contradicts the `require.True(t, ok)` assertion above because `ok` would only be true if there was a result, but `cs.Current` would be non-empty in that case. Also, instead of manually calling `Next`, can we use the [`requireCursorLength`](https://github.com/mongodb/mongo-go-driver/blob/9e2aca8afd8821e6b068cc2f25192bc640d90a0d/examples/documentation_examples/examples.go#L31) helper?

Reply:
Hmm, my understanding was that these lines were asserting that there _was_ a result (`require.True(t,ok)`), there was no error (`require.NoError(t, err)`), and that the number of documents still left in the cursor was **not** equal to 0 (`require.NotEqual(t, len(next), 0`). That seems consistent logically to me, but let me know if I misinterpreted one of those. 

I might end up switching to `requireCursorLength` one I put in some test data, but 2131 is checking if the cursor has a non-zero value of remaining documents, and `requireCursorLegnth

Reply:
I edited my previous comment. There's no contradiction here, but just checking that `Next` returns true is sufficient because the cursor API invariant is that `cursor.Current` will be a non-empty document if `Next` returns true.